docs/docs/guides/multi-stage-application-delivery.md

```diff
@@ -0,0 +1,340 @@
+---
+comments: true
+---
+
+# Multi-stage Application Delivery
+
+In this guide you will learn how to deploy
+an application across multiple stages while getting
+full observability of each of your workload deployments.
+This is shown based on an example where we deploy
+an application into two environments (`dev` and `production`),
+using [Argo CD](https://argo-cd.readthedocs.io/en/stable/).
+
+The promotion from `dev` to `production` is implemented using
+`promotionTasks` that can be executed after a successful
+application deployment.
+In this example, the promotion task creates a pull request
+to deploy the same workload version that has been deployed in the
+`dev` stage into the `production` stage.
+
+Using the observability capabilities of Keptn,
+we also add contextual metadata to each deployment,
+such as the commit ID which triggered the deployment of a new
+version.
+This metadata is added to the OpenTelemetry traces
+representing the deployment of an application version
+in a particular stage.
+Also, deployment traces for the `production` stage will
+contain references to the deployments of the `dev` stage,
+giving you full traceability of a workload version
+across different stages.
+
+## Before you begin
+
+This guide assumes the following:
+
+- Installed Keptn with the
+[promotionTasksEnabled flag](https://github.com/keptn/lifecycle-toolkit/blob/main/lifecycle-operator/chart/README.md#global)
+set to `true`.
+- Installed [Argo CD](https://argo-cd.readthedocs.io/en/stable/)
+- Followed the [instructions for setting up observability](./otel.md#requirements-for-opentelemetry)
+- Access to a GitHub repository for hosting your application manifests.
+- A Personal Access token for your GitHub repository.
+We recommend using a [fine-grained token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#fine-grained-personal-access-tokens)
+with the following permissions for the repository:
+    - Read access to metadata
+    - Read/Write access to actions
+
+Note that for the promotion to work properly,
+you need to enable read and write permissions for
+Github actions in your repository - see the screenshot below:
+
+![Workflow Permissions](./assets/multi-stage-delivery/workflow-permissions.png)
+
+## Preparing the Application Manifests
+
+In this example, the application will be deployed in
+two different namespaces, each representing a different
+stage (`dev` and `production`).
+To create the namespaces, execute the following commands:
```

Comment:
Should this be a numbered list using the imperative for each item?  If so, this would be the first item, "Create the namespaces by executing the following commands:"

This would get rid of all the "Next" items later on.